.github/workflows/pages.yml

```diff
@@ -0,0 +1,27 @@
+name: Github Pages
+
+on: [push]
+
+permissions:
+  contents: write
+
+jobs:
+  build-github-pages:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2 # repo checkout
+      - uses: actions-rs/toolchain@v1 # get rust toolchain for wasm
+        with:
+          toolchain: stable
+          target: wasm32-unknown-unknown
+          override: true
+      - name: Rust Cache # cache the rust build artefacts
+        uses: Swatinem/rust-cache@v1
+      - name: Downlaod and Install Trunk binary
+        run: wget -qO- https://github.com/thedodd/trunk/releases/latest/download/trunk-x86_64-unknown-linux-gnu.tar.gz | tar -xzf-
+      - name: Build # build
+        run: ./trunk build --release --public-url eframe_template # change this name to your project name
```

Comment:
> ./trunk build --release --public-url eframe_template

using `--public-url something` will allow trunk to modify all the href paths like from `favicon.ico` to `something/favicon.ico` . this is necessary for github pages where the site is deployed to `username.github.io/something` and all files must be requested relatively as `something/favicon.ico`. 

i have seen some people also use a "hack" by `sed` replacing all paths to use relative paths instead like `./favicon.ico` after trunk build step. this is used when you don't know which url your project is going to be deployed to.